roles/validate_inventory/tasks/day2.yml

```diff
@@ -3,3 +3,21 @@
     that:
       - day2_discovery_iso_name is defined
   when: (groups['day2_workers'] | default([])) | length > 0
+
+- name: Check for use_mce_for_day2 if required
+  assert:
+    that:
+      - use_mce_for_day2 is defined
+  when: (groups['day2_workers'] | default([])) | length > 0
```

Comment:
Put these all in a block instead copying the when :).

Reply:
done